src/popup/src/App.vue

```diff
@@ -28,7 +30,10 @@ export default defineComponent({
   watch: {
     showSnyk: {
       handler(showSnyk) {
-        chrome.storage.local.set({ showSnyk });
+        chrome.storage.local
+          .set({ showSnyk })
+          .then(() => chrome.storage.local.get())
+          .then(sendEventSettingsChanged);
```

Comment:
The *popup* is running on the **extension context**, and to pass the settings to the *injected script*, we need to send a message to the *content script*, which can send a `message` to the webpage (where the *injected script* is live).